src/main/java/frc/robot/Orchestrator.java

```diff
@@ -0,0 +1,287 @@
+import frc.robot.subsystems.indexer.IndexerConstants;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.pixy2.Pixy2;
+import frc.robot.subsystems.robotstate.RobotState;
+import frc.robot.subsystems.shooter.Shooter;
+import java.util.Set;
+import java.util.function.Supplier;
+
+public class Orchestrator {
+  private final Drive drive;
+  private final Intake intake;
+  private final Indexer indexer;
+  private final Shooter shooter;
+  private final Pixy2 pixy2;
+  private final Arm arm;
+  private final Translation2d speaker =
+      AllianceFlipUtil.apply(FieldConstants.Speaker.centerSpeakerOpening.toTranslation2d());
+
+  public Orchestrator(
+      Drive drive, Intake intake, Indexer indexer, Shooter shooter, Pixy2 pixy2, Arm arm) {
+    this.drive = drive;
+    this.intake = intake;
+    this.indexer = indexer;
+    this.shooter = shooter;
+    this.pixy2 = pixy2;
+    this.arm = arm;
+  }
+
+  /**
+   * Drives to one of the notes pre positioned on the field.
+   *
+   * @param targetTranslation The Translation2d for one of the notes on the field during auto.
+   * @return The command for driving to a note during auto.
+   */
+  public Command driveToNote(Translation2d targetTranslation) {
+    Supplier<Rotation2d> targetRotation =
+        () -> targetTranslation.minus(drive.getPose().getTranslation()).getAngle();
+    return Commands.defer(
+        () -> new StraightDriveToPose(new Pose2d(targetTranslation, targetRotation.get()), drive),
+        Set.of(drive));
+  }
+
+  /**
+   * Calculates the angle to turn to the speaker, and turns.
+   *
+   * @return The command for the robot to turn.
+   */
+  public Command turnToSpeaker() {
+    Supplier<Pose2d> targetPose =
+        () ->
+            new Pose2d(
+                drive.getPose().getTranslation(),
+                speaker
+                    .minus(drive.getPose().getTranslation())
+                    .getAngle()
+                    .minus(Rotation2d.fromDegrees(180)));
+    return Commands.defer(() -> new StraightDriveToPose(targetPose.get(), drive), Set.of(drive));
+  }
+
+  /**
+   * Same as shootBasic, but for scoring in the amp instead, so we are using a slower speed.
+   *
+   * @return The cammand to shoot into the amp.
+   */
+  public Command shootAmp() {
+    return Commands.sequence(
+        shooter.manualShoot(3.5).withTimeout(0.5),
+        Commands.parallel(indexer.setPercent(1), shooter.manualShoot(3.5))
+            .until(() -> !indexer.getLimitSwitchPressed())
+            .withTimeout(2));
+  }
+
+  /**
+   * Sets the arm to amp scoring position
+   *
+   * @return The command to move the arm to setpoint
+   */
+  public Command alignArmAmp() {
+    return arm.toSetpoint(ArmConstants.AMP_POSITION);
+  }
+
+  public Command goToAmp() {
+    Supplier<Pose2d> targetPose =
+        () ->
+            new Pose2d(
+                AllianceFlipUtil.apply(FieldConstants.ampCenter.getX()),
+                FieldConstants.ampCenter.getY() - 1,
```

Comment:
I assume `- 1` means that we're 1 meter away from the amp, right?

Move 1 and 0.5 below to constants or make them tunable.